File: app/src/main/java/com/widyatama/nurseassistant/features/jadwal/JadwalPresenter.java
package com.widyatama.nurseassistant.features.jadwal;

import com.widyatama.core.base.BasePresenter;
import com.widyatama.core.util.SchedulerProviderUtil;
import com.widyatama.nurseassistant.data.local.LocalStorageHelper;

import io.reactivex.disposables.CompositeDisposable;

public class JadwalPresenter<V extends JadwalViewContracts> extends BasePresenter<V> implements JadwalPresenterContracts<V> {

    LocalStorageHelper localStorage;
    SchedulerProviderUtil schedulerProvider;
    CompositeDisposable compositeDisposable;

    public JadwalPresenter(LocalStorageHelper localStorage, SchedulerProviderUtil schedulerProvider,
                           CompositeDisposable compositeDisposable){
        super(schedulerProvider, compositeDisposable);

        this.localStorage = localStorage;
        this.schedulerProvider = schedulerProvider;
        this.compositeDisposable = compositeDisposable;
    }

    @Override
    public void getJadwal(int limit) {
        compositeDisposable.add(localStorage.getSampleDatabase().jadwal().getAll()
            .compose(schedulerProvider.ioToMainFlowableScheduler())
            .subscribe(jadwals -> {
                view.showJadwal(jadwals);
            }, throwable -> {
                System.out.println("Error get jadwal");
            }));
    }
}

File: app/src/main/java/com/widyatama/nurseassistant/features/jadwal/JadwalPresenter.kt
package com.widyatama.nurseassistant.features.jadwal

import com.widyatama.core.base.BasePresenter
import com.widyatama.core.util.SchedulerProviderUtil
import com.widyatama.nurseassistant.data.local.LocalStorageHelper
import io.reactivex.disposables.CompositeDisposable

class JadwalPresenter<V: JadwalViewContracts> constructor(private var localStorage: LocalStorageHelper,
  schedulerProvider: SchedulerProviderUtil, compositeDisposable: CompositeDisposable)
    : BasePresenter<V>(schedulerProvider, compositeDisposable), JadwalPresenterContracts<V> {

    override fun getJadwal(limit: Int) {
        compositeDisposable.add(localStorage.sampleDatabase.jadwal().getAll()
                .compose(schedulerProvider.ioToMainFlowableScheduler())
                .subscribe({
                    view?.showJadwal(it)
                }, {
                    print("error get jadwal")
                }))
    }
}